o1vm/src/interpreters/mips/interpreter.rs

```diff
@@ -977,7 +982,27 @@ pub trait InterpreterEnv {
     fn request_hint_write(&mut self, addr: &Self::Variable, len: &Self::Variable);
 }
 
-pub fn interpret_instruction<Env: InterpreterEnv>(env: &mut Env, instr: Instruction) {
+pub fn interpret_instruction<
+    Env: InterpreterEnv<Position = crate::interpreters::mips::column::ColumnAlias>,
+>(
+    env: &mut Env,
+    instr: Instruction,
+) {
+    // We need to subtract the relations from the instruction usize.
+    env.activate_selector(usize::from(instr) - N_MIPS_REL_COLS);
+
+    // Assert that exactly one selector is active at a time.
+    env.add_constraint(
+        Instruction::iter().fold(Env::constant(0), |acc, x| {
+            acc + env.variable(x.to_column_alias())
+        }) - Env::constant(1),
+    );
+
+    // Assert all selectors are either zero or one.
```

Comment:
I would not add this in the interpreter, but instead outside, when getting the constraints.